File: Programmers/유석환/Q42578.swift
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var c = [String: Int]()
    
    for item in clothes {
        if c[item[1]] != nil {
            c[item[1]]! += 1
        } else {
            c[item[1]] = 2 // 착용하지 않는 경우 미리 포함
        }
    }
    
    var total = 1
    for item in c {
        total *= item.value
    }
    
    return total - 1
}

File: Programmers/노수일/Q42578.py
def solution(clothes):
    clothes_dict = {}
    for cloth in clothes:
        if cloth[1] in clothes_dict:
            clothes_dict[cloth[1]] += 1
        else:
            clothes_dict[cloth[1]] = 1

    answer = 1

    for key in clothes_dict:
        answer *= (clothes_dict[key] + 1)

    answer -= 1
    return answer


print(solution([["yellowhat", "headgear"], ["bluesunglasses", "eyewear"], ["green_turban", "headgear"], ["bluesunglasses", "face"]]))
print(solution([["crowmask", "face"], ["bluesunglasses", "face"], ["smoky_makeup", "face"]]))
